api/routes/job.py

```diff
@@ -41,20 +47,161 @@ def get_jobs(
     return s.JobOutList(jobs=cast(list, jobs))
 
 
-@job_router.post("/", status_code=status.HTTP_201_CREATED, response_model=s.JobOut)
+# user create job
+@job_router.post(
+    "/",
+    status_code=status.HTTP_201_CREATED,
+    response_model=s.JobOut,
+    responses={
+        status.HTTP_409_CONFLICT: {"description": "Selected service not found"},
+        status.HTTP_404_NOT_FOUND: {"description": "Selected file not found"},
+    },
+)
 def create_job(
     job: s.JobIn,
     db: Session = Depends(get_db),
     current_user: m.User = Depends(get_current_user),
 ):
+    """Creates new job"""
+
     new_job: m.Job = m.Job(
-        **job.model_dump(),
+        **job.model_dump(
+            exclude={
+                "lang",
+                "service_uuid",
+                "location_uuid",
+                "start_date",
+                "end_date",
+                "file_uuids",
+            }
+        ),
         owner_id=current_user.id,
+        start_date=datetime.fromisoformat(job.start_date) if job.start_date else None,
+        end_date=datetime.fromisoformat(job.end_date) if job.end_date else None,
     )
+
     db.add(new_job)
     db.commit()
-    log(log.INFO, "Created job [%s] for user [%s]", new_job.title, new_job.owner_id)
-    return new_job
+
+    if job.location_uuid:
+        location = db.scalar(sa.select(m.Location).where(m.Location.uuid == job.location_uuid))
+        if not location:
+            log(log.ERROR, "Location [%s] not found", job.location_uuid)
+            raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="Selected location not found")
+
+        new_job.location_id = location.id
+        log(log.INFO, "Location [%s] was added to job [%s]", location.id, new_job.id)
+
+    service = db.scalar(sa.select(m.Service).where(m.Service.uuid == job.service_uuid))
+    if not service:
+        log(log.ERROR, "Service [%s] not found", job.service_uuid)
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="Selected service not found")
```

Comment:
I think we need to perform these kind of check before we even start doing anything else. As far as I understand, there is no point in trying to create a job unless we do not have relevant service